fix: match migration guide examples between v5/v6, proof-read (#6676)
aws-amplify/docs: src/fragments/lib/troubleshooting/common/upgrading.mdx

## Step 2: Upgrade your Amplify project dependencies

<Callout>Make sure to delete your package-lock.json and node_modules folder before running `npm install`</Callout>

<BlockSwitcher>
  <Block name="Web">
    Upgrade to the latest Amplify library using the following command.

    ```
    npm install aws-amplify

>     ```

Reviewer: Think we need to add guidance on updating the version of `aws-amplify` to v6 in their _package.json_?

Author: Yeah, sounds good :). Do you recommend 'latest' or '^6.0.1'?

Reviewer 2: `^6.0.1` is probably more idiomatic

## Step 2: Upgrade your Amplify project dependencies

<Callout>Make sure to delete your package-lock.json and node_modules folder before running `npm install`</Callout>

<BlockSwitcher>
  <Block name="Web">
    Upgrade to the latest Amplify library using the following command.

    ```
    npm install aws-amplify
    ```

>     You can also upgrade by making sure the version of `aws-amplify` in your package.json file is `latest` or `^6.0.1` and re-installing dependencies by running `npm install`.

Reviewer: Not sure if "also" makes sense here. If a customer does not change the Amplify dependency version then whatever the current version they are pointing at would be installed?

Author: Actually no, if they use the above script `npm install aws-amplify` it will upgrade to the latest version. At least from my experimentation that is accurate.

## Step 2: Upgrade your Amplify project dependencies

<Callout>Make sure to delete your package-lock.json and node_modules folder before running `npm install`</Callout>

<BlockSwitcher>
  <Block name="Web">
    Upgrade to the latest Amplify library using the following command.

    ```
    npm install aws-amplify

>     ```

Reviewer: Thinking the install script should come after the version instructions?

Author: wondering if we use `npm install aws-amplify@latest` or `npm install aws-amplify>6` or something, if we could remove the addiitonal package.json instructions to keep things simple. Will POC.

Author: `npm install aws-amplify@latest` installs the latest version, pinning it in the package.json file to 6.0.8 currently.

Reviewer: Aligned with updating the script and removing line 59. For the script prefer `npm install aws-amplify@6` over `@latest` to avoid installing a different major version in the future

>     If you are using npm as a package manager, make sure the version of `aws-amplify` in your package.json file is `latest` or `^6.0.1` and re-installing dependencies by running `npm install`.

Reviewer: IIRC `6.0.1` had an issue and `6.0.2` was used as `latest` for the MV release (@jimblanc probably remembers better than me):
```suggestion
    If you are using npm as a package manager, make sure the version of `aws-amplify` in your package.json file is `latest` or `^6.0.2` and re-installing dependencies by running `npm install`.
```

Reviewer 2: Good catch, that's correct `6.0.2` was what we flipped `latest` to at launch.

Reviewer: @erinleigh90 the version here still needs to be updated 😅

    The `aws-amplify` package in v6 includes the categories below:

    - Auth
    - API
    - Storage
    - Analytics
    - DataStore
    - In-App Messaging

    If you previously listed `@aws-amplify` namespaced packages for the above categories as dependencies in your `package.json`, you will need to remove them to avoid dependency duplication.

    If you would like to use Geo, Predictions, PubSub, or Interactions you will need to install those packages separately. (see [category-specific migration instructions](/[platform]/build-a-backend/troubleshooting/migrate-from-javascript-v5-to-v6/#4-upgrade-category-usage))

  </Block>

  <Block name="React Native">

    In v6, the following packages have been replaced with `@aws-amplify/react-native` and should be removed from your `package.json`.

    * `amazon-cognito-identity-js`

>     * `aws-amplify-react-native`

Reviewer: Have we provided migration instructions for legacy React Native `Authenticator` customers to migrate to using the current `Authenticator` exported from `@aws-amplify/ui-react-native`? 

Also, afaik `@aws-amplify/react-native` is basically a net new package providing polyfills and while required for usage with JS v6, does not provide any of the esports previously available from `aws-amplify-react-native` (@HuiSF Can you please confirm 😅 ).

Reviewer 3: Good catch Caleb. `aws-amplify-react-native` exports all react-native UI components without any native modules. So it's not a replacement indeed.

However, `aws-amplify-react-native` source code is incompatible with Amplify JS v6, we may need to change the wording around removing this package while upgrading to v6.

Reviewer: > Have we provided migration instructions for legacy React Native Authenticator customers to migrate to using the current Authenticator exported from @aws-amplify/ui-react-native?

Just a heads that this PR does not need to include the React Native Authenticator migration, UI team will own it @erinleigh90

> However, aws-amplify-react-native source code is incompatible with Amplify JS v6, we may need to change the wording around removing this package while upgrading to v6.

Thinking that we should explicitly call out that `aws-amplify-react-native` is not supported in v6 and remove the language regarding dependency replacement here as there is no real correlation between `aws-amplify-react-native` and `@aws-amplify/react-native`. We can follow up with the legacy RN `Authenticator` migration guide as noted above

There does still need to be migration instructions for RN customers who are not using the `Authenticator`. Again would avoid verbiage that includes "replacement" since `@aws-amplify/react-native` is not a 1:1 replacement of `amazon-cognito-identity-js`, assuming we add a callout regarding the lack of support for `aws-amplify-react-native` above this block, suggest something along the lines of:

```suggestion
To upgrade React Native projects that do not utilize `aws-amplify-react-native`, remove `amazon-cognito-identity-js` from the project _package.json_ and install `@aws-amplify/react-native`:
```

And add an install script below:

```
npm install @aws-amplify/react-native
```

    Then upgrade/install the necessary dependencies using the following command:

    ```
    npm install aws-amplify @aws-amplify/react-native @react-native-community/netinfo @react-native-async-storage/async-storage react-native-get-random-values
    ```

    Note that v6 supports react-native v0.70+, so if you prefer manually upgrading dependencies double-check the version of react-native in your package.json file.

>     `@aws-amplify/react-native` requires a minimum iOS deployment target of `13.0`, if has not already been set to `13.01 or above, open the _Podfile_ located in the _ios_ directory and update the `target` value:

Reviewer: Think this typo may have been mine initially hehe:
```suggestion
    `@aws-amplify/react-native` requires a minimum iOS deployment target of `13.0`, if it has not already been set to `13.0` or above, open the _Podfile_ located in the _ios_ directory and update the `target` value:
```
